diktat-gradle-plugin/src/main/kotlin/com/saveourtool/diktat/plugin/gradle/Utils.kt

```diff
@@ -7,74 +7,18 @@
 package com.saveourtool.diktat.plugin.gradle
 
 import org.gradle.api.Project
-import java.io.File
-import java.nio.file.Files
-import java.nio.file.Path
+import org.gradle.api.file.RegularFile
+import org.gradle.api.provider.Provider
+import org.gradle.api.reporting.ReportingExtension
 
 /**
- * @param diktatExtension
- * @return returns sourceRootDir as projectDir for sarif report
+ * @param fileName
+ * @param extension
+ * @return default location of report with provided [extension]
  */
-fun Project.getSourceRootDir(diktatExtension: DiktatExtension): Path? = when {
-    diktatExtension.githubActions -> projectDir.toPath()
-    diktatExtension.reporter == "sarif" -> projectDir.toPath()
-    else -> null
-}
-
-/**
- * Create CLI flag to set reporter for ktlint based on [diktatExtension].
- * [DiktatExtension.githubActions] should have higher priority than a custom input.
- *
- * @param diktatExtension extension of type [DiktatExtension]
- * @return CLI flag as string
- */
-fun Project.getReporterType(diktatExtension: DiktatExtension): String {
-    val name = diktatExtension.reporter.trim()
-    val validReporters = listOf("sarif", "plain", "json", "html")
-    val reporterType = when {
-        diktatExtension.githubActions -> {
-            if (diktatExtension.reporter.isNotEmpty()) {
-                logger.warn("`diktat.githubActions` is set to true, so custom reporter [$name] will be ignored and SARIF reporter will be used")
-            }
-            "sarif"
-        }
-        name.isEmpty() -> {
-            logger.info("Reporter name was not set. Using 'plain' reporter")
-            "plain"
-        }
-        name !in validReporters -> {
-            logger.warn("Reporter name is invalid (provided value: [$name]). Falling back to 'plain' reporter")
-            "plain"
-        }
-        else -> name
-    }
-
-    return reporterType
-}
-
-/**
- * Get destination file for Diktat report or null if stdout is used.
- * [DiktatExtension.githubActions] should have higher priority than a custom input.
- *
- * @param diktatExtension extension of type [DiktatExtension]
- * @return destination [File] or null if stdout is used
- */
-internal fun Project.getOutputFile(diktatExtension: DiktatExtension): File? = when {
-    diktatExtension.githubActions -> project.layout.buildDirectory
-        .file("reports/diktat/diktat.sarif")
-        .get()
-        .asFile
-        .also {
-            Files.createDirectories(it.parentFile.toPath())
-        }
-    diktatExtension.output.isNotEmpty() -> file(diktatExtension.output)
-    else -> null
-}
-
-/**
- * Whether SARIF reporter is enabled or not
- *
- * @param reporterFlag
- * @return whether SARIF reporter is enabled
- */
-internal fun isSarifReporterActive(reporterFlag: String) = reporterFlag.contains("sarif")
+internal fun Project.defaultReportLocation(
+    extension: String,
```

Comment:
actually, we don't use this method for plain reported, since we usually use stdout for plain reporter.
also, I'm not sure about such extension for plain, I guess it should be `txt`